### PR TITLE
weakens medipens in outside of lavaland alt to 51706

### PR DESCRIPTION
**Pull request by Tlaltecuhtli**
:cl: MRP agent
balance: survival medipen heals normal on lavaland and less on station +  small side effects
balance: lavaland extract (one of the chem in the medipens) doesnt heal and does dizzyness  
and 15% prob of vomit on station
balance: lavaland extract OD does only 3 tox dmg when on station
/:cl:

- the lavaland extract (one of the healing chem in medipen with biggest healing power) doesnt heal on station and gives you shaky screen with chance of vomiting, making it usable in emergency sitations but not that great in middle of a fight or to spam heal while being beaten by people,
if you syringe it out then you will still get the chance of vomit and lose part of your chem + stun so its fine

- lore 

closes: #51706

### CODE REVIEW

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 1]
```diff
@@ -1019,13 +1019,19 @@
...
 
 /datum/reagent/medicine/lavaland_extract/on_mob_life(mob/living/carbon/M)
-	M.heal_bodypart_damage(5,5)
+	var/turf/T = get_turf(M)
+	if(is_mining_level(T.z))
+		M.heal_bodypart_damage(5,5)
+	else
+		M.Dizzy(10)
+		if(prob(15))
+			M.vomit()
```
Comment (Time-Green) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm:1034]: You sneakily never actually heal them on station and instead just make them constantly vomit and dizzy without any redeeming qualities. Brilliant.

**Comment by Time-Green** [comment]
> making it usable in emergency sitations but not that great in the middle of a fight or to spam heal while being beaten by people

Am I missing something? This is completely unusable in both scenarious, since it will never heal off-station

**Comment by Time-Green** [comment]
Hell, this is an even bigger nerf. In the other PR you just couldn't use it off-station; in this one it lets you use it but just completely wrecks you without healing.

**Comment by Tlaltecuhtli** [comment]
vomit is a stun and its bad while being in a fight, it does heal from the other chems which have their own limits (oxal and salacid heal only under 50 hp)

**Comment by Time-Green** [comment]
Ah, the injector has multiple chems. I thought I was going mad

**Comment by necromanceranne** [comment]
So if I syringe out a ton of lavaland extract, I have a potent poison which I can also deliver via medipen applicator.

**Comment by KathyRyals** [comment]
I agree that lavaland extract is not the biggest healer in the pen. Also in my PR you can still create situations where you can use the pen on-station. Blow up a room, fight in space etc... And it fits the niche of "Don't fight a miner in a low pressure environment due to their weapons" more.

**Comment by ATH1909** [comment]
> So if I syringe out a ton of lavaland extract, I have a potent poison which I can also deliver via medipen applicator.

a 15% chance to stun every 2 seconds (that purges some of the chems in your system) isn't THAT good

like, people aren't using stuff like liquid electricity in their kill mixes currently, right?

**Comment by XDTM** [comment]
Lavaland extract heals 5 brute and burn per tick so i'd argue it is the best healing reagent in the pen. Plus, this PR makes a better parallel to PKAs, since they don't arbitrarily lock up in high pressure, they just deal less damage. The medipen here would still be a cocktail of heals, but you'd miss out of a lot of healing and risk a stun as a side effect, so it's decent for healing but not for combat, so it does have on-station uses.

As for the poison comment, is vomit that powerful of an effect?

**Comment by ATH1909** [comment]
> I agree that lavaland extract is not the biggest healer in the pen. Also in my PR you can still create situations where you can use the pen on-station. Blow up a room, fight in space etc... And it fits the niche of "Don't fight a miner in a low pressure environment due to their weapons" more.

aye, but the vomiting is gonna purge some of the other, better healing chems in the pen

**Comment by ATH1909** [comment]
by the way, what happens if you take this chem on station without a stomach? do you just have to deal with the dizziness?

**Comment by XDTM** [comment]
Even if you're immune to vomiting it should be fine. It is still a limited medical item, after all, so it's fair that it does do some healing.

**Comment by kriskog** [comment]
> 
> 
> So if I syringe out a ton of lavaland extract, I have a potent poison which I can also deliver via medipen applicator.

Yes, self-purging on stun poison. How devious.

File: code/modules/reagents/reagent_containers/hypospray.dm [hunk 2]
```diff
@@ -215,7 +215,7 @@
 
 /obj/item/reagent_containers/hypospray/medipen/survival
 	name = "survival medipen"
-	desc = "A medipen for surviving in the harshest of environments, heals and protects from environmental hazards. WARNING: Do not inject more than one pen in quick succession."
+	desc = "A medipen for surviving in the harshest of environments, heals and protects from environmental hazards. WARNING: Do not inject more than one pen in quick succession, the Nanotransen Exotic Medical Research Division is not responsible for any side effects caused by usage in untested enviroments like space stations."
```
Comment (Ziiro) [inline code/modules/reagents/reagent_containers/hypospray.dm:218]: I wonder if, for fluff/lore reasons, you should say it's something gravity related. "Intended for use on planetary bodies. May cause unintended side effects under the effects of zero or artificial gravity"

Then again this is another can of worms with Ice Box, so I don't know.

**Comment by nightred** [comment]
Would it not make more sense to do this?
```
M.heal_bodypart_damage(5,5)
if(!is_mining_level(T.z))
    M.Dizzy(10)```

**Comment by Tlaltecuhtli** [comment]
but it will do the heal that way

File: code/modules/reagents/reagent_containers/hypospray.dm (hunk 2 above)
Comment (Tlaltecuhtli) [reply code/modules/reagents/reagent_containers/hypospray.dm:218]: lore on why is written in lavaland extract desc

**Comment by Kelenius** [comment]
So this turns the medipen into a very powerful weapon on the station?

**Comment by nightred** [comment]
> but it will do the heal that way

Heal and has side effects, your way makes it a weapon.

**Comment by Tlaltecuhtli** [comment]
if you get medipenned twice/ 3 times there wont be much difference from before as the extract OD will be countered by sal / oxal under 50hp
 please no more dumb comments

**Comment by necromanceranne** [comment]
Okay, so it took me a bit to work out the big issue here. What you are actually doing is making lavaland extract one of maybe the scarier poisons crew can get. And an applicator that bypasses hardsuits no less.

The only remaining chems in a survival medipen that actually heals is omnizine (5u), salacid (8u) and oxad (8u), which in general is really quite pathetic for an actual combat heal for station combat unless you're using it to counteract some other minor damage from another chem. Which means actually just going up and stabbing a guy with two pens is going to inject them with enough lavaland extract to overdose, doing 3 burn, brute and toxin (2 burn and brute due to the other healing chems), and eventually stun. 

But you could bypass that entirely, get yourself the pens, a syringe (cargo can print them), chemmaster (condimasters work fine here too), a dart gun of some kind (bamboo works, traitors get one out of the uplink that is also silenced), remove the extract, and you now have maybe the most lethal poison in the game if your overdose someone (3u or more, so you get two overdoses per three pens, 1500 points, super cheap). If they don't vomit, they're melting. If they do vomit, they're stunned for 8 seconds. A win-win. Histamine doesn't even get this good while overdosing, and you only need small doses for this to work since you don't have the liver toxin purge limitation holding you back. So long as you can inject that lethal threshold, the poor sap you shot is gonna die.

You can combine this with some chem that removes nutrients from their body to also boost the stun to 20 seconds if you get it in time. Great.

You also seem to be aware of the fact you can take out the chem from survival medipens but because you can't write to save your life, it's hard to parse what you actually mean. Do you mean taking out your poison, or taking out the healing chems? Quite honestly, if this is the state you leave every PR body, you should just have it closed immediately, good lord dude, I can barely work out what your intent even is meant to be.

**Comment by XDTM** [comment]
that's easily solved by making the overdose also limitec to lavaland

**Comment by optimumtact** [comment]
changelog

**Comment by Tlaltecuhtli** [comment]
the effort you would have to put to do that you could just get the super stun mushrooms from lavaland but i guess i ll do the xdtm suggestion anyway

**Comment by KathyRyals** [comment]
Since you didn't answer when I asked, maybe because you knew it wouldn't work, I went ahead and tested your changes.

Like I suspected, you can use the pens just fine in IceBox, no nausea in the station.

Also worth noting that on other maps where you get the nausea effect, I jabbed myself with three pens and didn't puke ONCE, just got some screen shake. Now, I know this is hardly a good sample size, but it feels like the chance to puke is too weak to be relevant, and in the end, it's still a magic injector pen that heals (slightly less) on station.

Also I'll say it one more time : this PR is scuffed because you can use the pens just fine on station if the map is IceBox.
image

**Comment by KathyRyals** [comment]
GIF of pen usage on icebox :


ezgif com-optimize

**Comment by Tlaltecuhtli** [comment]
if the ice box z level is set to mining z level thats a ice box problem

**Comment by KathyRyals** [comment]
That's not a good solution because there can be fauna on icebox z level iirc

**Comment by Doosteroni** [comment]
> That's not a good solution because there can be fauna on icebox z level if I remember correctly.

All it is are a bunch of harmless wolves if you have a pen you shouldn’t be on the station z-level anyways. I still don’t get the purpose of this nerf. It seems like a solution to a problem that _doesn’t exist_

**Comment by XDTM** [comment]
I haven't played on icebox yet, is it meant to replace lavaland or is it still separate? The z-level tags should reflect that appropriately. If they're separate, then maybe the flag should be split into Planet and Lavaland.

In case lavaland ends up on the same z-level as the station, the check can be switched to atmospheric pressure instead, but it would be kinda weird as the user could be isolated from ambient pressure by spacesuits.

**Comment by Tlaltecuhtli** [comment]
if miners mine in the really small ice zone around station they get weaker medipens which should be fine for the weak mobs like  single wolves, as the big bad things cant spawn here ( bear caves, portals, megafauna etc) so its fine

**Comment by KathyRyals** [comment]
I still don't think that's a big enough nerf. Yes they don't get the effect of the lavaland extract, and they get a TINY chance to throw up, but they still have full effect of all the other chems in the pens, which heals for A LOT.

**Comment by Tlaltecuhtli** [comment]
15% chance each tick isnt a tiny chance, its enough to make the user think if getting the stun is worth the 8u of sal acid and oxand,

the other medipen chems are all available to crew by medkits from medical, chem dispenser, cargo and even mining vendor can sell you a standard brute medkit with salcyclic acid medipen, i dont intend to remove 100% of the medical power of the medipen used on station as thats dumb and doesnt make sense, i m putting the survival medipen to the same level of on station meds and removing any abusive usage like medipenning and face tanking shit like blob

**Comment by KathyRyals** [comment]
It's like you didn't even look at the pen. It contains 10u Salbutamol, 15u Leporazine, 10u Epinephrine, 2u Lavaland Extract, 5u Omnizine, 8u Oxandrolone, 8u Salicyclic Acid and 2u Morphine.  You realize no other items on station can inject THAT much  healing chems in one instant?? Also you say you can fin all those chems on station. First of all, that's not entirely true. You can't find five units of omnizine that easily, and this couple with all the other chems, with ONE injection ? That's literally the best healing pen on station. Your 15% chance to throw up is gone after like 5 ticks ? (2u * 0.4u per tick). That's barely a nerf.

**Comment by Tlaltecuhtli** [comment]
the only real healing chems are extract, salcyclic and oxandrolone, the rest is just filler

**Comment by KathyRyals** [comment]
Omnizine.

**Comment by SkeletalElite** [comment]
> Omnizine.

Didn't omnizine get mega nerfed and it barely heals anything anymore?

**Comment by KathyRyals** [comment]
5u of omnizine takes 50 ticks to metabolize. Which heals for 25 damage of each type. That's still pretty sizeable.

**Comment by Shadowflame909** [comment]
> 5u of omnizine takes 50 ticks to metabolize. Which heals for 25 damage of each type. That's still pretty sizeable.

Still leaves people coming to the MD for tend wounds. Which is your overall goal.

Thus, balanced.

**Comment by ExcessiveUseOfCobblestone** [comment]
no it doesnt, it gets them coming for another pen which is why you're so desperate to write paragraphs on every PR touching it.

**Comment by Shadowflame909** [comment]
I'm desperate to not have them nerfed everywhere because I can't play lavaland without them.

I approve of this nerf and any nerf that makes them less accessible on station. Heck if they blew up on station and you could only leave them at lavaland I'd be fine with that.

Needless to say the person I was commenting against is a medbay main and has bias about these pens taking their job.

**Comment by KathyRyals** [comment]
That's absolutely stupid. It's like you didn't even make an effort to understand or read what I said. I'm not complaining about the pens taking medical's job. I want the pen to be nerfed because of antags/validhunters abusing them to become invincible killing machines on station. Which is why I think this PR is not the solution : it makes the healing barely weaker

**Comment by Shadowflame909** [comment]
Your goals are very noble and are quite logical.

But mining is one of the only jobs I play. It is an entirely different balanced game then on station. Everything brings your health to near-crit within a few seconds.

Even one of the most "buff-up" jobs like xenobio, can barely assist in easing the difficulty.

The reason for this is that one of the lavaland coders designed it with such separation that lavaland only items would ONLY be helpful on lavaland. Which even nullifies the majority of the lavaland loot, because they're a bit more then useless on lavaland (like the katana and ERT hardsuit) but quite powerful on station.

Your viewpoint is shared and is accurately reflected in one of the design goals of the headcoder. Preventing player interaction with lavaland, and limiting it's interaction with off-station mechs.

So, I feel the best way to go about this is to not have any PRs that focus on merging the two and allowing the items to be able to interact across the z-levels.

Lavaland Only Items are the goal forward, and future nerfs would be best advised to reflect this. It keeps both jobs intact, and doesn't just create a bandaid that will ripped off and trampled on in the short future.
